### ISSUE TITLE
Testing mega-issue

### ISSUE THREAD

**Issue by ebsmothers** [pytorch/torchtune#691]
Creating this issue to track gaps in our current testing.

### Tests to write
- [ ] Add gradient accumulation test for LoRA recipe (ideally also testing LR scheduler)
- [ ] Add tests for Mistral models
- [ ] Add tests for Gemma models
- [ ] Add DPO recipe test
- [ ] Add test for DPOLoss
- [ ] Add CI for 13B QLoRA
- [ ] Testing of external APIs

### Quality-of-life changes
- [ ] Mock datasets locally for recipe tests
- [ ] Nightly regression test CI job
- [ ] Move away from /tmp/ folders
